pkg/build/api/validation/validation.go

```diff
@@ -23,11 +34,20 @@ func ValidateBuildConfig(config *buildapi.BuildConfig) errs.ValidationErrorList
 	allErrs := errs.ValidationErrorList{}
 	if len(config.Name) == 0 {
 		allErrs = append(allErrs, errs.NewFieldRequired("name", config.Name))
+	} else if !util.IsDNSSubdomain(config.Name) {
+		allErrs = append(allErrs, errs.NewFieldInvalid("name", config.Name, "name must be a valid subdomain"))
```

Comment:
nothing in that says name fields have to be dns subdomains as I read it:

name: a string that uniquely identifies this object within the current namespace (see identifiers.md). This value is used in the path when retrieving an individual object.

Name : A non-empty string guaranteed to be unique within a given scope at a particular time; used in resource URLs; provided by clients at creation time and encouraged to be human friendly; intended to facilitate creation idempotence and space-uniqueness of singleton objects, distinguish distinct entities, and reference particular entities across operations.